frame/offences/src/lib.rs

```diff
@@ -216,28 +299,45 @@ struct TriageOutcome<T: Config> {
 #[must_use = "The changes are not saved without called `save`"]
 struct ReportIndexStorage<T: Config, O: Offence<T::IdentificationTuple>> {
 	opaque_time_slot: OpaqueTimeSlot,
+	session_index: SessionIndex,
 	concurrent_reports: Vec<ReportIdOf<T>>,
+	session_reports: Vec<(Kind, OpaqueTimeSlot, ReportIdOf<T>)>,
 	_phantom: PhantomData<O>,
 }
 
 impl<T: Config, O: Offence<T::IdentificationTuple>> ReportIndexStorage<T, O> {
 	/// Preload indexes from the storage for the specific `time_slot` and the kind of the offence.
-	fn load(time_slot: &O::TimeSlot) -> Self {
+	fn load(time_slot: &O::TimeSlot, session_index: SessionIndex) -> Self {
```

Comment:
won't this need any sort of migration for the existing offences, adding `session_index` and `session_reports` to this struct, consequently populating `SessionReports`?

Reply:
We would only be adding new offences to the `SessionReports`. Once we are past the `MaxSessionReportAge`, we can just clean up the old data via a migration?

Reply:
`ReportIndexStorage` is a shim and not persisted to storage itself, so i think its fine.